optax/_src/transform.py

```diff
@@ -869,6 +869,64 @@ def update_fn(updates, state, params=None):
   return base.GradientTransformation(init_fn, update_fn)
 
 
+class ScaleByNewtonState(NamedTuple):
+  """State holding the inverse of the sum of gradient squares to date."""
+
+  hessian_inv: base.Updates
+
+
+def sherman_morrison(a_inv, u, v):
+  den = 1.0 + (u @ a_inv) @ v
+  a_inv -= a_inv @ jnp.outer(u, v) @ a_inv / den
+  return a_inv
+
+
+def scale_by_newton(eps: float = 1e-7) -> base.GradientTransformation:
+  """Rescale updates by multiplying by the inverse of an approximation of the
+  hessian.
+
+  References
```

Comment:
As above, please format this similar to e.g. `rmsprop` in `alias.py` disabling and re-enabling the `line-too-long` warning if necessary.